src/odb/src/codeGenerator/schema/chip/dbMarkerCategory.json

```diff
@@ -0,0 +1,27 @@
+{
+  "name":"dbMarkerCategory",
+  "type":"dbObject",
+  "fields":[
+    {
+      "name":"_name",
+      "type":"char *",
+      "default": "nullptr",
+      "flags":["no-set"]
+    },
+    {
+      "name":"description_",
+      "type":"std::string"
+    },
+    {
+      "name":"source_",
+      "type":"std::string",
+      "flags":["no-get"]
+    },
+    {
+      "name": "max_markers_",
+      "type": "int",
+      "default": "10000"
```

Comment:
What is the purpose of this?

Reply:
to provide a mechanism to limit the number of markers that can get created. I suppose the default could be unlimited and then allow the tools to place an upper bound. (It was an attempt to have something standard across the different tools).

Reply:
Is there any user control or support for it in the tools or is it for future use?

Reply:
The category stops creating dbMarkers when the limit is reached. So the enforcement is in odb. Each tool is already checking if the marker returned from `dbMarker::create` is a nullptr before populating it with information.